File: Hash Table/0720.Longest Word in Dictionary/Solution.java
class Solution {
    public String longestWord(String[] words) {
        String ans = "";
        Set<String> set = new HashSet<>(Arrays.asList(words));
        for (String s : set) {
            int n = s.length(), m = ans.length();
            if (n < m) {
                continue;
            }
            if (n == m && s.compareTo(ans) > 0) {
                continue;
            }
            boolean flag = true;
            for (int i = 1; i <= n; i++) {
                String sub = s.substring(0, i);
                if (!set.contains(sub)) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                ans = s;
            }
        }
        return ans;
    }
}

File: Hash Table/0720.Longest Word in Dictionary/Solution.py
class Solution:
    def longestWord(self, words: List[str]) -> str:
        ans = ""
        w_set = set(words)
        for s in w_set:
            n, m = len(s), len(ans)
            if n < m:
                continue
            if n == m and s > ans:
                continue
            flag = True
            for i in range(1, n + 1):
                sub = s[:i]
                if sub not in w_set:
                    flag = False
                    break
            if flag:
                ans = s
        return ans